Description: Check the result of is_distro_package() and fix the check if the
 kernel header package is installed at all.
Author: Felix Geyer <debfx@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/940285

--- dkms-2.2.0.3.orig/dkms_apport.py
+++ dkms-2.2.0.3/dkms_apport.py
@@ -48,12 +48,16 @@ if options.kernel:
     # TODO: Ubuntu specific
     kernel_package = "linux-headers-" + options.kernel
 
+    supported_kernel = True
     try:
-        apport.packaging.is_distro_package(kernel_package)
+        supported_kernel = apport.packaging.is_distro_package(kernel_package)
     except ValueError, e:
-        if e == 'package does not exist':
-            print >> sys.stderr, 'ERROR (dkms apport): kernel package %s is not supported' % (kernel_package)
-            sys.exit(1)
+        if str(e) == 'package does not exist':
+            supported_kernel = False
+
+    if not supported_kernel:
+        print >> sys.stderr, 'ERROR (dkms apport): kernel package %s is not supported' % (kernel_package)
+        sys.exit(1)
 
 make_log=os.path.join('/var','lib','dkms',options.module,options.version,'build','make.log')
 
